testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/TestContainersEnricher.java

```diff
@@ -0,0 +1,69 @@
+package org.wildfly.test.integration.observability;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.arquillian.test.spi.TestEnricher;
+import org.jboss.as.test.shared.util.AssumeTestGroupUtil;
+import org.wildfly.test.integration.observability.container.OpenTelemetryCollectorContainer;
+
+public class TestContainersEnricher implements TestEnricher {
+    private static final ThreadLocal<OpenTelemetryCollectorContainer> otelCollectorContainer = new ThreadLocal<>();
+
+    @Override
+    public void enrich(Object testCase) {
+        if (AssumeTestGroupUtil.isDockerAvailable()) {
+            for (Field field : getFieldsWithAnnotation(testCase.getClass())) {
+                OpenTelemetryCollectorContainer value = lookup(field.getType());
+                try {
+                    if (!field.canAccess(testCase)) {
+                        field.setAccessible(true);
+                    }
+                    field.set(testCase, value);
+                } catch (Exception e) {
+                    throw new RuntimeException("Could not set value on field " + field + " using " + value, e);
+                }
+            }
+        }
+    }
+
+    private OpenTelemetryCollectorContainer lookup(Class<?> type) {
+        if (type.isAssignableFrom(OpenTelemetryCollectorContainer.class)) {
+
+            OpenTelemetryCollectorContainer container = otelCollectorContainer.get();
+            if (container == null) {
+                container = new OpenTelemetryCollectorContainer();
+                container.start();
+                otelCollectorContainer.set(container);
+            }
+            return container;
+        }
+        return null;
+    }
+
+    @Override
+    public Object[] resolve(Method method) {
+        return new Object[0];
```

Comment:
This violates the (unclear) TestEnricher contract, although the caller (LocalTestExecuter) handles implementations doing this instead of returning null.

Reply:
Strange. Pretty sure I ran into trouble returning null there, and aped that approach from somewhere, but that was before PTO. I changed that to null, and it seems to be working as expected.